cmd/kube-apiserver/app/server.go

```diff
@@ -238,19 +238,39 @@ func (s *APIServer) Run(_ []string) error {
 		glog.Fatalf("Failure to start kubelet client: %v", err)
 	}
 
-	disableV1beta3 := false
-	v1beta3FlagValue, ok := s.RuntimeConfig["api/v1beta3"]
-	if ok && v1beta3FlagValue == "false" {
-		disableV1beta3 = true
+	// "api/all=false" allows users to selectively enable specific api versions.
+	disableAllAPIs := false
+	allAPIFlagValue, ok := s.RuntimeConfig["api/all"]
```

Comment:
I had to go look this thing up. Is it documented anywhere? I would greatly prefer not having magic keys like "api/all" or "api/legacy"; I'd prefer an opt-out or opt-in list, and clarification (in the command line flag help message!) that it determines which api versions are *served* (as opposed to the other env var which determines which are *understood*).

I guess you didn't add api/legacy, so let's not touch that one. But can we avoid adding api/all?